x/txfees/keeper/feedecorator.go

```diff
@@ -348,3 +358,31 @@ func DeductFees(txFeesKeeper types.TxFeesKeeper, bankKeeper types.BankKeeper, ct
 
 	return nil
 }
+
+// CalculatePriorityFromFees calculates the priority of a transaction based on the fees paid.
+// Higher fees result in higher priority. The priority is calculated by converting the fee to base denom
+// and using its amount as the priority value.
+func (dfd DeductFeeDecorator) CalculatePriorityFromFees(ctx sdk.Context, fees sdk.Coins) (int64, error) {
+	if fees.IsZero() {
+		return 0, nil
+	}
+
+	// Convert fee to base denom for consistent priority calculation
+	baseDenom, err := dfd.txFeesKeeper.GetBaseDenom(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	// If fee is already in base denom, use its amount directly
+	if fees[0].Denom == baseDenom {
+		return fees[0].Amount.Int64(), nil
```

Comment:
Can this overflow?

Reply:
Good shout, I can write a test for this :bow: